PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,423 @@
+    if ($null -ne $Haschildren) {
+        $Fix= "FIX --> Please follow the following article https://aka.ms/setpf to move $($PublicFolderInfo.Publicfolder.Identity) dumpster children found to under NON_IPM_SUBTREE scope e.g. \NON_IPM_SUBTREE\DUMPSTER_ROOT\DUMPSTER_EXTEND\RESERVED_1\RESERVED_1"
+        $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) dumpster has $($Haschildren.Name.count) subfolder(s) which is a blocker for deletion operations over the public folder!`nDumpster subfolder(s) found:`n----------------------------`n$($Haschildren.identity)"
+        WritetoScreenANDlog -Issue $Issue -Fix $Fix
+    }
+}
+Function ValidateMEPFGuid {
+    param([PSCustomObject]$PublicFolderInfo)
+    #validate if MailRecipientGuid parameter is found empty/null
+    if ($PublicFolderInfo.Publicfolder.MailEnabled -eq $true) {
+        $Hasvalue=$PublicFolderInfo.Publicfolder.MailRecipientGuid.Guid
+        if ($null -eq $Hasvalue -or $Hasvalue -eq "00000000-0000-0000-0000-000000000000") {
+            $Fix= "FIX --> Please follow the following article https://aka.ms/EnableMPF to mail enable the affected public folder to generate a GUID over MailRecipientGuid parameter, validate MailRecipientGuid parameter has a GUID using https://aka.ms/getpf article then mail disable the affected public folder back again using https://aka.ms/disablempf article to be able to remove the public folder as requested."
+            $Issue="Mail-enabled public folder $($PublicFolderInfo.Publicfolder.Identity) is unhealthy e.g MailRecipientGuid parameter is found empty/null which is a blocker for deletion operations over the public folder"
+            WritetoScreenANDlogDlog -Issue $Issue -Fix $Fix
+        }
+    }
+}
+Function ExtractLog {
+    param([PSCustomObject]$PublicFolderInfo)
+    #create zip file for logs folder
+    $tstamp= Get-Date -Format yyyyMMdd_HHmmss
+    if (!(Test-Path  "$ExportPath\logs_$tstamp")) {
+        mkdir "$ExportPath\logs_$tstamp" -Force | Out-Null
+    }
+    $PublicFolderInfo.Publicfolder | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\Publicfolder.txt" -NoClobber
+    $PublicFolderInfo.Publicfolderdumpster | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\Publicfolderdumpster.txt" -NoClobber
+    $PublicFolderInfo.pfmbx | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\pfmbx.txt" -NoClobber
+    $PublicFolderInfo.PfMBXstats | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\PfMBXstats.txt" -NoClobber
+    $PublicFolderInfo.IPM_SUBTREE | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\IPM_SUBTREE.txt" -NoClobber
+    $PublicFolderInfo.NON_IPM_SUBTREE | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\NON_IPM_SUBTREE.txt" -NoClobber
+    $PublicFolderInfo.DUMPSTER_ROOT | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\DUMPSTER_ROOT.txt" -NoClobber
```

Comment:
Exporting out to txt only is a horrible idea. Export out to xml makes it so all data will be there and is not truncated.

Reply:
Done.